packages/ui/components/ui/tx/actions-views/ibc-relay.tsx

```diff
@@ -0,0 +1,141 @@
+import { ViewBox } from '../viewbox';
+import { ActionDetails } from './action-details';
+import {
+  FungibleTokenPacketData,
+  IbcRelay,
+} from '@penumbra-zone/protobuf/penumbra/core/component/ibc/v1/ibc_pb';
+import { MsgRecvPacket } from '@penumbra-zone/protobuf/ibc/core/channel/v1/tx_pb';
+import { MsgUpdateClient } from '@penumbra-zone/protobuf/ibc/core/client/v1/tx_pb';
+import { UnimplementedView } from './unimplemented-view.tsx';
+import { uint8ArrayToBase64 } from '@penumbra-zone/types/base64';
+import { getUtcTime } from './isc20-withdrawal.tsx';
+import { useMemo } from 'react';
+
+// Packet data stored as json string encoded into bytes
+const parsePacket = ({ packet }: MsgRecvPacket): FungibleTokenPacketData | undefined => {
+  if (!packet?.data) {
+    return undefined;
+  }
+
+  try {
+    const dataString = new TextDecoder().decode(packet.data);
+    return FungibleTokenPacketData.fromJsonString(dataString);
+  } catch (e) {
+    return undefined;
```

Comment:
Suggestion: should we error-log here, or display something to the user even?

Reply:
Hmm, perhaps we do. I'll handle the case where the packet cannot be parsed and display `Unknown packet data` in the row.